site/pages/api/create-consequence-operator.test.ts

```diff
@@ -0,0 +1,111 @@
+        const { req, res } = getMockRequestAndResponse({ body: {}, mockWriteHeadFn: writeHeadMock });
+        createConsequenceOperator(req, res);
+
+        const errors: ErrorInfo[] = [
+            { errorMessage: "Select at least one operator", id: "consequenceOperator" },
+            { errorMessage: "Enter a consequence description", id: "description" },
+            { errorMessage: "Enter planned or unplanned", id: "removeFromJourneyPlanners" },
+            { errorMessage: "Select the severity from the dropdown", id: "disruptionSeverity" },
+            { errorMessage: "Select atleast one direction", id: "disruptionDirection" },
+        ];
+        expect(setCookieOnResponseObject).toHaveBeenCalledTimes(1);
+        expect(setCookieOnResponseObject).toHaveBeenCalledWith(
+            COOKIES_CONSEQUENCE_OPERATOR_ERRORS,
+            JSON.stringify({ inputs: req.body, errors }),
+            res,
+        );
+        expect(writeHeadMock).toBeCalledWith(302, { Location: CREATE_CONSEQUENCE_OPERATOR_PATH });
+    });
+
+    it("should redirect back to /create-consequence-operator when description is too long", () => {
+        const operatorData: ConsequenceOperatorPageInputs = {
+            ...defaultOperatorData,
+            description:
+                "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
+        };
+
+        const { req, res } = getMockRequestAndResponse({ body: operatorData, mockWriteHeadFn: writeHeadMock });
+
+        createConsequenceOperator(req, res);
+
+        const errors: ErrorInfo[] = [{ errorMessage: "Description must not exceed 500 characters", id: "description" }];
+        expect(setCookieOnResponseObject).toHaveBeenCalledTimes(1);
+        expect(setCookieOnResponseObject).toHaveBeenCalledWith(
+            COOKIES_CONSEQUENCE_OPERATOR_ERRORS,
+            JSON.stringify({ inputs: req.body, errors }),
+            res,
+        );
+        expect(writeHeadMock).toBeCalledWith(302, { Location: CREATE_CONSEQUENCE_OPERATOR_PATH });
+    });
+
+    it("should redirect back to /create-consequence-operator when invalid time is passed", () => {
+        const operatorData: ConsequenceOperatorPageInputs = {
+            ...defaultOperatorData,
+            disruptionDelay: "7280",
```

Comment:
Im not sure why this is an invalid input - wouldnt this mean 72 hours and 80 minutes?

Reply:
I understand 80 minutes is 1 hour and 20 minutes, but im not sure this is invalid, and if it is im not sure the error message we give the user will be very clear.

Reply:
The regex pattern verifies that the last two characters entered are valid minute digits. Along with this the pattern ensures that the user doesn't enter random numbers such as 1234567. The regex pattern also limits the length of the input. The non specific error message covers these validations.

If specific errors are to be mentioned then additional validations will need to be added

Reply:
We're asking for the user to prove hours and minutes, in hhmm format. 80 is a number of minutes, because we arent asking for a time. We're asking for an amount of time.

We either need to be allowing 7280, or giving the user a better understanding of why 7280 is not a valid input.

Reply:
@alvak1 have you spoke to Steve or anybody about what we expect the users to enter? And what we should allow?

Reply:
@RobCattonKPMG  spoke to Steve and was informed that the design will be updated to "Enter time in minutes" with limit of 999. Update the code to reflect the same. minor changes to TimeSelector components added

Reply:
@alvak1 ok good stuff